[terra-modal-manager] Changed modal name to be the same as h1 title of the modal dialog (#1646)
cerner/terra-framework: packages/terra-abstract-modal/CHANGELOG.md

## Unreleased

>   * Changed
>     * Updated to read the title of the modal by the screen reader when a modal is opened.

Reviewer: ```suggestion
  * Added
    * Added screen reader support to read the title of the modal when it is open.
```
